### ISSUE TITLE
Feature Request: Babel 7.x support

### ISSUE THREAD

**Issue by fbartho** [commenthol/named-regexp-groups#1]
I just integrated this library last week and was really finding it useful. Unfortunately, this week when I upgraded to babel 7.0 (because react-native also upgraded), I ended up getting this error when running named-regexp-groups in my react-native project: 

>Couldn't find preset "es2015" relative to directory

I was able to work around this by patching `node_modules/named-regexp-groups/package.json` so that `babel.presets = []` in a post install script!

I'd be happy to contribute a Pull-Request that updates this to babel-7, or to help out in some other way.

**Comment by commenthol** [comment]
Ok did not know that Babel7 requires such changes. As you have already found a solution a PR is very welcome.

**Comment by commenthol** [comment]
Your observed issue made me curious. Maybe not following the [recommended](https://github.com/rollup/rollup/wiki/pkg.module)
way to handle `pkg.module` exports may cause this issue.
To correct this, I updated the package to export ES5 as recommended usign rollup as bundler.
Would you mind to try out this version `1.0.3-0` and let me know if your usecase works with the changes?

**Comment by fbartho** [comment]
@commenthol -- I tried with 1.0.3-0, and unfortunately, (nearly) the same error occurs. I also had to change my workaround patch to package.json to delete the env section when I'm working around it 😉 

The error is now:
```
error: bundling failed: Error: Couldn't find preset "env" relative to directory "/Users/fbarthelemy/Code/TRClient/node_modules/named-regexp-groups"
```

I think this would be resolvable if we make sure that babel-preset-env was a direct dependency, or a peer dependency of named-regex-groups. -- I'm no babel expert, so I don't know if that's how we're supposed to resolve it. But if I include:
```
		"babel-preset-env": "^1.6.1",
		"babel-preset-stage-2": "^6.24.1",
```
in my app's dependencies, the problem also goes away.

**Comment by commenthol** [comment]
I have the feeling this is related to the "babel" config inside package.json. Just published a new version v1.0.3-1 which has separate "babel" config in ".babelrc" which is not bundled inside the published version.

**Comment by fbartho** [comment]
@commenthol Success!!

**Comment by commenthol** [comment]
cool

**Comment by commenthol** [comment]
just published v1.0.3
